thesis/Chapters/Chapter2.tex

```diff
@@ -0,0 +1,105 @@
+\end{itemize}
+
+
+\subsection{Storybook}
+
+Stampix is using Storybook\footnote{https://storybook.js.org/} to create a component library for the mobile application. With Storybook, the operations team can change some parameters and then export a configuration.
+This is nice because Storybook provides a sort of sandbox to play around and design a deployment, completely separated from any cloud resources.
+
+Problem with this is that Storybook focuses mostly on styling. We need more than just styling.
+
+%----------------------------------------------------------------------------------------
+
+\section{Automated testing}
+
+We create lots of different webapps, a different one for each client. Currently, we have no automated tests. This should change
+
+Automated testing gives developers confidence their changes did not break anything. It can spot bugs before the code is even released
+
+A lot of the following tools do very similar things, they emulate or run a browser and we can instruct it to do certain actions on our site after which we can assert if the application is behaving correctly.
+
+
+
+// TODO: Since most of these tools are/do basically the same thing, we should def focus on the differences between them.
+
+\subsection{Selenium}
+
+Selenium\footnote{https://www.selenium.dev/} is an established project
+
+\subsection{Cypress}
+
+Cypress\footnote{https://www.cypress.io/} is pretty new and 'cool' 
+
+\subsection{Protractor}
+
+\subsection{Playwright}
+
+Playwright\footnote{https://playwright.dev/} is
+
+%----------------------------------------------------------------------------------------
+
+\section{Deployment}
+
+We will be deploying lots of apps for different customers. It's important that we can do this as rapidly and effortlessly as possible.
+This means we should adopt infrastructure as code as much as possible. There's a number of tools available, I will talk about some of the big players.
+
+\subsection{Ansible}
+
+Ansible\footnote{https://www.ansible.com/} is an agentless automation tool. Ansible connects via SSH to a number of hosts you specify and executes tasks on those hosts. 
+It offers a range of powerful tools and a strong ecosystem. Ansible focuses mostly on configuration management.
+
+While there are a few community and official AWS-related plugins available, I don't think Ansible is a good tool for this project. 
+Ansible is built around SSH connections and our system will have (practically) no SSH-capable machines.
+
+\subsection{Terraform}
+
+Terraform \footnote{https://www.terraform.io/} is a infrastructure as code tool. In contrast to Ansible, 
+Terraform focuses mostly on infrastructure tasks like creating virtual machines or allocating cloud resources.
+
+\subsection{Serverless}
+\subsection{AWS Cloud Development Kit}
+\subsection{Cloudformation}
+
+// Cloudformation is kind of a legacy thing vs AWS CDK but probably we should still include it?
```

Comment:
The CDK still transpiles to Cloudformation (or Terraform), CDK is just an abstraction layer